src/js_native_api_v8.cc

```diff
@@ -335,6 +336,32 @@ class Reference : private Finalizer, RefTracker {
   bool _delete_self;
 };
 
+class ArrayBufferReference final : public Reference {
+ public:
+  // Same signatures for ctor and New() as Reference:
+  template <typename... Args>
+  explicit ArrayBufferReference(Args&&... args)
```

Comment:
Nit: not sure if applicable but could we perhaps add a bit of type safety:
```C++
  explicit ArrayBufferReference(napi_env env,
                                v8::Local<v8::ArrayBuffer> value,
                                Args&&... args)
```

Reply:
@lundibundi Done, PTAL :+1: